app/controllers/application_controller/filter.rb

```diff
@@ -1,10 +1,13 @@
 # Filter/search/expression methods included in application.rb
+require_relative '../../helpers/miq_expression/filter_subst'
```

Comment:
This should not be be necessary if the file is named properly and the class matches. On that note, I am just confused about what the new module class is (since there's no other usage of it), and thus am not sure where it *should* live.  The `helpers` are actually "view helpers", so if this is not used in a view, then it doesn't belong here.  If it's a controller mixin, then it should live in controllers/mixins, however the OP states that you want to use it in automate.

To me this just feels like it should be a model class that you just instantiate and use or perhaps as a part of MiqExpression the model.  Can you elaborate a bit on how this helps you?